### PR TITLE
[#19][UI] As a user, I can submit the survey

### LINKED ISSUE #19: [UI] As a user, I can submit the survey
## Why
The application supports various types of survey items. After the user fills in all required survey items, the survey can be submitted and mark the survey as completed.

## Acceptance Criteria
Display submit button on the last survey item.
image

### PR DESCRIPTION
**Pull request by hanam1ni**
- Closes #19 

## What happened 👀

- Submit survey responses to the API.
- When submitting invalid survey responses, display an error message. 
 
## Proof Of Work 📹

- Display an error message when survey responses are invalid

image

- Survey is submitted when survey responses are valid

https://user-images.githubusercontent.com/14077479/178776581-4e24903d-5604-41e1-a9d5-0972a101bcc7.mov

### CODE REVIEW

File: src/pages/surveys/[id]/questions/[currentOrder].page.tsx [hunk 1]
```diff
@@ -35,8 +40,15 @@ const Question: NextPage = () => {
   ) as SurveyQuestionInterface;
 
   const [isLoading, setIsLoading] = useState(true);
+  const [surveySubmitError, setSurveySubmitError] = useState<string[]>([]);
```
Comment (malparty) [inline src/pages/surveys/[id]/questions/[currentOrder].page.tsx:43]:
```suggestion
  const [surveySubmitErrors, setSurveySubmitErrors] = useState<string[]>([]);
```
😇

Comment (malparty) [reply src/pages/surveys/[id]/questions/[currentOrder].page.tsx:43]: After reading the whole file, it seems like you do not need an array here... 🤔 
What about using a string and renaming it to "ErrorMessage"? 

```suggestion
  const [submitErrorMessage, setSubmitErrorMessage] = useState('');
```

File: src/pages/surveys/[id]/questions/[currentOrder].page.tsx [hunk 2]
```diff
@@ -90,9 +140,23 @@ const Question: NextPage = () => {
...
-            <Button label="Submit" className="px-8" />
+            <Button
+              label="Submit"
+              className="px-8"
+              onClick={onSubmitSurvey}
+              disabled={isEmptySurveyResponse || isSurveySubmit}
+            />
           )}
         </div>
+        {surveySubmitError.length !== 0 && (
```
Comment (malparty) [inline src/pages/surveys/[id]/questions/[currentOrder].page.tsx:151]:
```suggestion
        {submitErrorMessage && (
```
As strings aggregates to `false` when empty, the suggestion bellow makes this line easier to understand too :)

**Review by malparty — APPROVED** [review]
A small suggestion but no blockers ;-) 

Approved ahead! :)

File: src/pages/surveys/[id]/questions/[currentOrder].page.tsx (hunk 2 above)
Comment (hanam1ni) [reply src/pages/surveys/[id]/questions/[currentOrder].page.tsx]: Updated in [3d184a9](https://github.com/hanam1ni/react-surveyor/pull/89/commits/3d184a9c26fcb80a4da2c9876b79e9f6d350c8ba)

**Review by malparty — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**